dist/s_comment.py

```diff
@@ -0,0 +1,44 @@
+import sys
+
+comment = str()
+in_multiline = False
+line_length = 100
+indentation = 0
+
+for line in sys.stdin:
+    sline = line.strip()
+    # Beginning of a multi-line comment.
+    if sline.startswith('/*') and not sline.endswith('*/'):
+        assert not in_multiline
+        in_multiline = True
+        # Figure out how far we need to indent.
+        for c in line:
+            if c == ' ':
+                indentation += 1
+            else:
+                break
+    # End of a comment. If we were in the middle of a multi-line comment then
+    # write it.
+    elif sline.endswith('*/'):
```

Comment:
No need to look at this file yet. It's on the right track but needs a bit of refinement and cleanup.